core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/RepositoryTransactionContext.java

```diff
@@ -280,7 +280,7 @@ public Long commit(String author, String commitComment, String parentLockContext
 			log().info("Persisting changes to {}@{}", path(), timestamp);
 			commit = staging.commit(null, timestamp, author, commitComment);
 			log().info("Changes have been successfully persisted to {}@{}.", path(), timestamp);
-			return commit == null ? null : commit.getTimestamp();
+			return commit == null ? Commit.NO_COMMIT_TIMESTAMP : commit.getTimestamp();
```

Comment:
It turns out there is exactly one caller of `TransactionContext#commit` that is interested in the timestamp, returned via a `CommitResult` instance: https://github.com/b2ihealthcare/snow-owl/blob/7f5e60d95cfba13301a842b43f5455660d2197e0/snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java#L216